### PR TITLE
Even better size optimisations

### PR DESCRIPTION
**Pull request by nvuillam**
- Remove `.npm/_cacache`
- Industrialize rust packages installation in one block, then remove cache

### CODE REVIEW

**Comment by nvuillam** [comment]
## [🦙 MegaLinter](https://oxsecurity.github.io/megalinter/beta) status: ❌ [**ERROR**](https://github.com/oxsecurity/megalinter/actions/runs/3032690972)

|    Descriptor    |                                                        Linter                                                        |Files|Fixed|                                  Errors                                   |Elapsed time|
|------------------|----------------------------------------------------------------------------------------------------------------------|-----|----:|---------------------------------------------------------------------------|------------|
|✅ BASH           |[bash-exec](https://oxsecurity.github.io/megalinter/beta/descriptors/bash_bash_exec)                                  |    7|     |                                                                          0|0.02s       |
|✅ BASH           |[shellcheck](https://oxsecurity.github.io/megalinter/beta/descriptors/bash_shellcheck)                                |    7|     |                                                                          0|0.26s       |
|✅ BASH           |[shfmt](https://oxsecurity.github.io/megalinter/beta/descriptors/bash_shfmt)                                          |    7|    0|                                                                          0|0.12s       |
|✅ COPYPASTE      |[jscpd](https://oxsecurity.github.io/megalinter/beta/descriptors/copypaste_jscpd)                                     |yes  |     |no                                                                         |2.69s       |
|✅ DOCKERFILE     |[hadolint](https://oxsecurity.github.io/megalinter/beta/descriptors/dockerfile_hadolint)                              |  104|     |                                                                          0|7.81s       |
|✅ JSON           |[eslint-plugin-jsonc](https://oxsecurity.github.io/megalinter/beta/descriptors/json_eslint_plugin_jsonc)              |   21|    0|                                                                          0|2.52s       |
|✅ JSON           |[jsonlint](https://oxsecurity.github.io/megalinter/beta/descriptors/json_jsonlint)                                    |   19|     |                                                                          0|0.23s       |
|✅ JSON           |[v8r](https://oxsecurity.github.io/megalinter/beta/descriptors/json_v8r)                                              |   21|     |                                                                          0|13.15s      |
|:warning: MARKDOWN|[markdownlint](https://oxsecurity.github.io/megalinter/beta/descriptors/markdown_markdownlint)                        |  292|    0|[**5**](https://github.com/oxsecurity/megalinter/actions/runs/3032690972)  |5.17s       |
|✅ MARKDOWN       |[markdown-link-check](https://oxsecurity.github.io/megalinter/beta/descriptors/markdown_markdown_link_check)          |  292|     |                                                                          0|4.6s        |
|✅ MARKDOWN       |[markdown-table-formatter](https://oxsecurity.github.io/megalinter/beta/descriptors/markdown_markdown_table_formatter)|  292|    0|                                                                          0|6.56s       |
|✅ OPENAPI        |[spectral](https://oxsecurity.github.io/megalinter/beta/descriptors/openapi_spectral)                                 |    1|     |                                                                          0|0.77s       |
|:warning: PYTHON  |[bandit](https://oxsecurity.github.io/megalinter/beta/descriptors/python_bandit)                                      |  169|     |[**55**](https://github.com/oxsecurity/megalinter/actions/runs/3032690972) |2.58s       |
|✅ PYTHON         |[black](https://oxsecurity.github.io/megalinter/beta/descriptors/python_black)                                        |  169|    1|                                                                          0|5.43s       |
|✅ PYTHON         |[flake8](https://oxsecurity.github.io/megalinter/beta/descriptors/python_flake8)                                      |  169|     |                                                                          0|2.02s       |
|✅ PYTHON         |[isort](https://oxsecurity.github.io/megalinter/beta/descriptors/python_isort)                                        |  169|    1|                                                                          0|0.49s       |
|✅ PYTHON         |[mypy](https://oxsecurity.github.io/megalinter/beta/descriptors/python_mypy)                                          |  169|     |                                                                          0|16.33s      |
|✅ PYTHON         |[pylint](https://oxsecurity.github.io/megalinter/beta/descriptors/python_pylint)                                      |  169|     |                                                                          0|11.58s      |
|:warning: PYTHON  |[pyright](https://oxsecurity.github.io/megalinter/beta/descriptors/python_pyright)                                    |  169|     |[**276**](https://github.com/oxsecurity/megalinter/actions/runs/3032690972)|16.82s      |
|✅ REPOSITORY     |[checkov](https://oxsecurity.github.io/megalinter/beta/descriptors/repository_checkov)                                |yes  |     |no                                                                         |16.28s      |
|✅ REPOSITORY     |[git_diff](https://oxsecurity.github.io/megalinter/beta/descriptors/repository_git_diff)                              |yes  |     |no                                                                         |0.13s       |
|✅ REPOSITORY     |[secretlint](https://oxsecurity.github.io/megalinter/beta/descriptors/repository_secretlint)                          |yes  |     |no                                                                         |8.07s       |
|✅ REPOSITORY     |[trivy](https://oxsecurity.github.io/megalinter/beta/descriptors/repository_trivy)                                    |yes  |     |no                                                                         |20.64s      |
|❌ SPELL          |[cspell](https://oxsecurity.github.io/megalinter/beta/descriptors/spell_cspell)                                       |  701|     |[**9**](https://github.com/oxsecurity/megalinter/actions/runs/3032690972)  |18.57s      |
|✅ SPELL          |[misspell](https://oxsecurity.github.io/megalinter/beta/descriptors/spell_misspell)                                   |  530|    0|                                                                          0|0.5s        |
|✅ XML            |[xmllint](https://oxsecurity.github.io/megalinter/beta/descriptors/xml_xmllint)                                       |    3|     |                                                                          0|0.0s        |
|✅ YAML           |[prettier](https://oxsecurity.github.io/megalinter/beta/descriptors/yaml_prettier)                                    |   79|    0|                                                                          0|2.77s       |
|✅ YAML           |[v8r](https://oxsecurity.github.io/megalinter/beta/descriptors/yaml_v8r)                                              |   22|     |                                                                          0|32.01s      |
|✅ YAML           |[yamllint](https://oxsecurity.github.io/megalinter/beta/descriptors/yaml_yamllint)                                    |   80|     |                                                                          0|1.23s       |

See errors details in [**artifact MegaLinter reports** on CI Job page](https://github.com/oxsecurity/megalinter/actions/runs/3032690972)

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: flavors/documentation/action.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ outputs:
     description: "0 if no source file has been updated, 1 if source files has been updated"
 runs:
   using: "docker"
-  image: "docker://oxsecurity/megalinter-documentation:v6.9.1"
+  image: "docker://oxsecurity/megalinter-documentation:v6"
```
Comment (echoix) [inline flavors/documentation/action.yml:10]: Is that the expected output for action files?

**Review by echoix — COMMENTED** [review]
How does GitHub actions versioning work? Does this mean it overrides the original v6 release?

File: flavors/documentation/action.yml (hunk 1 above)
Comment (nvuillam) [reply flavors/documentation/action.yml:10]: GitHub actions looks for the branch/tag in the related repo, then it uses the related action.yml

When a new MegaLinter release is uploaded, it overrides "v6" image so v6 is always the latest v6 minor/patch version

**Comment by nvuillam** [comment]
## [🦙 MegaLinter](https://oxsecurity.github.io/megalinter/test-nuillam-more) status: :warning: [**WARNING**](https://github.com/oxsecurity/megalinter/actions/runs/3037847437)

|     Descriptor     |                                                              Linter                                                               |Files|Fixed|                                  Errors                                   |Elapsed time|
|--------------------|-----------------------------------------------------------------------------------------------------------------------------------|-----|----:|---------------------------------------------------------------------------|------------|
|✅ BASH             |[bash-exec](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/bash_bash_exec)                                  |    7|     |                                                                          0|0.01s       |
|✅ BASH             |[shellcheck](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/bash_shellcheck)                                |    7|     |                                                                          0|0.3s        |
|✅ BASH             |[shfmt](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/bash_shfmt)                                          |    7|    0|                                                                          0|0.05s       |
|✅ COPYPASTE        |[jscpd](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/copypaste_jscpd)                                     |yes  |     |no                                                                         |3.23s       |
|✅ DOCKERFILE       |[hadolint](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/dockerfile_hadolint)                              |  104|     |                                                                          0|7.86s       |
|✅ JSON             |[eslint-plugin-jsonc](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/json_eslint_plugin_jsonc)              |   21|    0|                                                                          0|2.37s       |
|✅ JSON             |[jsonlint](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/json_jsonlint)                                    |   19|     |                                                                          0|0.26s       |
|✅ JSON             |[v8r](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/json_v8r)                                              |   21|     |                                                                          0|14.35s      |
|:warning: MARKDOWN  |[markdownlint](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/markdown_markdownlint)                        |  292|    2|[**5**](https://github.com/oxsecurity/megalinter/actions/runs/3037847437)  |5.94s       |
|✅ MARKDOWN         |[markdown-link-check](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/markdown_markdown_link_check)          |  292|     |                                                                          0|5.36s       |
|✅ MARKDOWN         |[markdown-table-formatter](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/markdown_markdown_table_formatter)|  292|    2|                                                                          0|6.73s       |
|✅ OPENAPI          |[spectral](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/openapi_spectral)                                 |    1|     |                                                                          0|0.93s       |
|:warning: PYTHON    |[bandit](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/python_bandit)                                      |  169|     |[**55**](https://github.com/oxsecurity/megalinter/actions/runs/3037847437) |2.92s       |
|✅ PYTHON           |[black](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/python_black)                                        |  169|    0|                                                                          0|4.48s       |
|✅ PYTHON           |[flake8](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/python_flake8)                                      |  169|     |                                                                          0|2.57s       |
|✅ PYTHON           |[isort](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/python_isort)                                        |  169|    0|                                                                          0|0.51s       |
|✅ PYTHON           |[mypy](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/python_mypy)                                          |  169|     |                                                                          0|23.48s      |
|✅ PYTHON           |[pylint](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/python_pylint)                                      |  169|     |                                                                          0|14.42s      |
|:warning: PYTHON    |[pyright](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/python_pyright)                                    |  169|     |[**274**](https://github.com/oxsecurity/megalinter/actions/runs/3037847437)|22.09s      |
|✅ REPOSITORY       |[checkov](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/repository_checkov)                                |yes  |     |no                                                                         |18.86s      |
|:warning: REPOSITORY|[devskim](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/repository_devskim)                                |yes  |     |[**60**](https://github.com/oxsecurity/megalinter/actions/runs/3037847437) |1.32s       |
|✅ REPOSITORY       |[dustilock](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/repository_dustilock)                            |yes  |     |no                                                                         |2.36s       |
|✅ REPOSITORY       |[git_diff](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/repository_git_diff)                              |yes  |     |no                                                                         |0.04s       |
|✅ REPOSITORY       |[secretlint](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/repository_secretlint)                          |yes  |     |no                                                                         |4.39s       |
|✅ REPOSITORY       |[syft](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/repository_syft)                                      |yes  |     |no                                                                         |2.18s       |
|✅ REPOSITORY       |[trivy](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/repository_trivy)                                    |yes  |     |no                                                                         |21.27s      |
|✅ SPELL            |[cspell](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/spell_cspell)                                       |  701|     |                                                                          0|25.98s      |
|✅ SPELL            |[misspell](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/spell_misspell)                                   |  530|    2|                                                                          0|0.49s       |
|✅ XML              |[xmllint](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/xml_xmllint)                                       |    3|     |                                                                          0|0.0s        |
|✅ YAML             |[prettier](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/yaml_prettier)                                    |   79|    0|                                                                          0|3.17s       |
|✅ YAML             |[v8r](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/yaml_v8r)                                              |   22|     |                                                                          0|35.4s       |
|✅ YAML             |[yamllint](https://oxsecurity.github.io/megalinter/test-nuillam-more/descriptors/yaml_yamllint)                                    |   80|     |                                                                          0|1.57s       |

See errors details in [**artifact MegaLinter reports** on CI Job page](https://github.com/oxsecurity/megalinter/actions/runs/3037847437)

You could have the same capabilities but better runtime performances if you use a MegaLinter flavor:
- [**oxsecurity/megalinter/flavors/python@test-nvuillam-more**](https://oxsecurity.github.io/megalinter/test-nuillam-more/flavors/python/) (56 linters)

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

**Pull request merged**